runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -0,0 +1,372 @@
+package org.corfudb.runtime.view;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.recovery.FastObjectLoader;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.clients.IClientRouter;
+import org.corfudb.runtime.clients.LayoutClient;
+import org.corfudb.runtime.clients.LogUnitClient;
+import org.corfudb.runtime.clients.ManagementClient;
+import org.corfudb.runtime.exceptions.AlreadyBootstrappedException;
+import org.corfudb.runtime.exceptions.LayoutModificationException;
+import org.corfudb.runtime.exceptions.OutrankedException;
+import org.corfudb.runtime.exceptions.QuorumUnreachableException;
+import org.corfudb.runtime.exceptions.RecoveryException;
+
+/**
+ * A view of the Layout Manager to manage reconfigurations of the Corfu Cluster.
+ *
+ * <p>Created by zlokhandwala on 11/1/17.</p>
+ */
+@Slf4j
+public class LayoutManagementView extends AbstractView {
+
+    public LayoutManagementView(@NonNull CorfuRuntime runtime) {
+        super(runtime);
+    }
+
+    private volatile long prepareRank = 1L;
+
+    /**
+     * Recover cluster from layout.
+     *
+     * @param recoveryLayout Layout to use to recover
+     * @return True if the cluster was recovered, False otherwise
+     */
+    public void recoverCluster(Layout recoveryLayout)
+            throws QuorumUnreachableException, OutrankedException, InterruptedException,
+            ExecutionException, LayoutModificationException, CloneNotSupportedException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.corfudb.runtime.exceptions.LayoutModificationException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)